angular/src/app/views/settings/settings.component.ts

```diff
@@ -99,17 +110,45 @@ export class SettingsComponent {
     );
   }
 
+  private loadOrderStatusValues(): void {}
+
   /**
    * Load the settings in an Object into the in-memory settings
    * @param data the settings as an Object
    */
-  private loadSettingsFromObject(data): void {
+  private loadSettingsFromObject(data: SettingsInterface): void {
     this.stockBuffer = data.stockBuffer;
-    this.defaultOrderStatus = data.defaultOrderStatus;
     this.defaultShippingProvider = data.defaultShippingProvider;
     this.defaultItemMappingMethod = data.defaultItemMappingMethod;
     this.importOrdersSince = data.importOrdersSince;
     this.isAllInventorySyncEnabled = data.isAllInventorySyncEnabled;
+
+    this.chooseOrderStatusAfterRefreshingList(data.defaultOrderStatus);
+  }
+
+  private chooseOrderStatusAfterRefreshingList(statusId): void {
+    this.orderStatusService.fetch().subscribe(
+      (data) => {
+        // store this so that the UI can show in drop-down
+        this.orderStatuses = data;
+        if (statusId) {
+          if (this.orderStatuses.length > 0) {
+            for (let option of this.orderStatuses) {
+              if (option.statusId == statusId) {
+                this.defaultOrderStatus = statusId;
+                return;
+              }
```

Comment:
If `this.orderStatuses` is a list, then I think here you can use the `.forEach()` method to set the orderStatus .
```
this.orderStatuses.forEach(option => {
  this.defaultOrderStatus = statusId
})
```

Reply:
In this case we're just trying to loop over the array, and if we find an element that has the ID that the user picked, we want to **exit the function** and avoid falling-back to the default. The forEach in javascript takes a callback as its argument, and a **return** would just return from the callback, not from the outer method.

Reply:
Ok got it